### PR TITLE
migrate logs to structed logging

### PR DESCRIPTION
**Pull request by fancc**
Migrate some logs to structured logging

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Ref:

[keps/sig-instrumentation/1602-structured-logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)
[Structured Logging migration instructions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md)

**Does this PR introduce a user-facing change?:**:
```
Migrate some log messages to structured logging
````

### CODE REVIEW

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 			if attempt > 1 {
 				newResources = GetDeletableResources(discoveryClient)
 				if len(newResources) == 0 {
-					klog.V(2).Infof("no resources reported by discovery (attempt %d)", attempt)
+					klog.V(2).InfoS("no resources reported by discovery", "Attempt", attempt)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:208]:
```suggestion
					klog.V(2).InfoS("No resources reported by discovery", "Attempt", attempt)
```

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 2]
```diff
@@ -318,13 +318,13 @@ func (gc *GarbageCollector) attemptToDeleteWorker() bool {
 		if !existsInGraph {
 			// this can happen if attemptToDelete loops on a requeued virtual node because attemptToDeleteItem returned an error,
 			// and in the meantime a deletion of the real object associated with that uid was observed
-			klog.V(5).Infof("item %s no longer in the graph, skipping attemptToDeleteItem", n)
+			klog.V(5).InfoS("item no longer in the graph, skipping attemptToDeleteItem", "item", n)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:321]:
```suggestion
			klog.V(5).InfoS("Item no longer in the graph, skipping attemptToDeleteItem", "item", n)
```

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 3]
```diff
@@ -318,13 +318,13 @@ func (gc *GarbageCollector) attemptToDeleteWorker() bool {
...
 			return true
 		}
 		if nodeFromGraph.isObserved() {
 			// this can happen if attemptToDelete loops on a requeued virtual node because attemptToDeleteItem returned an error,
 			// and in the meantime the real object associated with that uid was observed
-			klog.V(5).Infof("item %s no longer virtual in the graph, skipping attemptToDeleteItem on virtual node", n)
+			klog.V(5).InfoS("item no longer virtual in the graph, skipping attemptToDeleteItem on virtual node", "item", n)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:327]:
```suggestion
			klog.V(5).InfoS("Item no longer virtual in the graph, skipping attemptToDeleteItem on virtual node", "item", n)
```

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 4]
```diff
@@ -345,7 +345,7 @@ func (gc *GarbageCollector) attemptToDeleteWorker() bool {
 			//    have a way to distinguish this from a valid type we will recognize
 			//    after the next discovery sync.
 			// For now, record the error and retry.
-			klog.V(5).Infof("error syncing item %s: %v", n, err)
+			klog.V(5).InfoS("error syncing item", "item", n, "error", err)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:348]:
```suggestion
			klog.V(5).InfoS("Error syncing item", "item", n, "err", err)
```

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 5]
```diff
@@ -355,7 +355,7 @@ func (gc *GarbageCollector) attemptToDeleteWorker() bool {
 		// requeue if item hasn't been observed via an informer event yet.
 		// otherwise a virtual node for an item added AND removed during watch reestablishment can get stuck in the graph and never removed.
 		// see https://issue.k8s.io/56121
-		klog.V(5).Infof("item %s hasn't been observed via informer yet", n.identity)
+		klog.V(5).InfoS("item hasn't been observed via informer yet", "item", n.identity)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:358]:
```suggestion
		klog.V(5).InfoS("Item hasn't been observed via informer yet", "item", n.identity)
```

File: pkg/controller/garbagecollector/garbagecollector.go [hunk 6]
```diff
@@ -587,7 +587,7 @@ func (gc *GarbageCollector) attemptToDeleteItem(item *node) error {
 func (gc *GarbageCollector) processDeletingDependentsItem(item *node) error {
 	blockingDependents := item.blockingDependents()
 	if len(blockingDependents) == 0 {
-		klog.V(2).Infof("remove DeleteDependents finalizer for item %s", item.identity)
+		klog.V(2).InfoS("remove DeleteDependents finalizer for item", "item", item.identity)
```
Comment (yangjunmyfm192085) [inline pkg/controller/garbagecollector/garbagecollector.go:590]:
```suggestion
		klog.V(2).InfoS("Remove DeleteDependents finalizer for item", "item", item.identity)
```

**Comment by caesarxuchao** [comment]
/ok-to-test

The suggested changes from @yangjunmyfm192085 look good. Please apply. Thank you both.

**Comment by yangjunmyfm192085** [comment]
duplicated with https://github.com/kubernetes/kubernetes/pull/100821
/wg structured-logging

**Pull request closed without merging**